frontend/src/CardGallery/components/NavBar.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import MenuIcon from '@mui/icons-material/Menu';
+import {Menu, MenuItem} from "@mui/material";
+import {NavLink} from "react-router-dom";
+import "../components/css/NavBar.css"
+import {AccountCircle} from "@mui/icons-material";
+
+
+export default function NavBar() {
+
+    const [menu, setMenu] = React.useState<null | HTMLElement>(null)
+    const [userLogin, setUserLogin] = React.useState<null | HTMLElement>(null)
+
+    const open = Boolean(menu)
+    
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setMenu(event.currentTarget);
+    }
+
+    const handleClose = () => {
+        setMenu(null)
+        setUserLogin(null)
+    }
+
+    const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
+        setUserLogin(event.currentTarget);
+
+    };
+
+
+    return (
+        <Box sx={{ flexGrow: 1 }}>
+            <AppBar position="static">
+                <Toolbar>
+                    <IconButton
+                        size="large"
+                        edge="start"
+                        color="inherit"
+                        aria-label="menu"
+                        sx={{ mr: 2 }}
+                        id="basic-button"
+                        aria-controls={open ? 'basic-menu' : undefined}
+                        aria-haspopup="true"
+                        aria-expanded={open ? 'true' : undefined}
+                        onClick={handleClick}
```

Comment:
`onClick={handleClick}` You could be a bit more creative with the naming what's actually happening here.